test/test-config.sh

```diff
@@ -162,9 +162,9 @@ fi
 # EndpointSlice is disabled because of https://github.com/kubernetes/kubernetes/issues/91287 (Kubernetes
 # < 1.19) and because there were random connection failures to node ports during sanity
 # testing (Kubernetes 1.19.0)
-: ${TEST_FEATURE_GATES:=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true,CSIInlineVolume=true\
-$(case ${TEST_KUBERNETES_VERSION} in 1.1[0-5]) ;; *) echo ',EndpointSlice=false';; esac)\
-$(case ${TEST_KUBERNETES_VERSION} in 1.19) echo ',CSIStorageCapacity=true,GenericEphemeralVolume=true,EndpointSliceProxying=false';; esac)\
+: ${TEST_FEATURE_GATES:=CSINodeInfo=true,CSIDriverRegistry=true,CSIBlockVolume=true,CSIInlineVolume=true,\
+$(case ${TEST_KUBERNETES_VERSION} in 1.1[6-9]) echo 'EndpointSlice=false,EndpointSliceProxying=false,';; esac)\
+$(case ${TEST_KUBERNETES_VERSION} in 1.19 | 1.20) echo 'CSIStorageCapacity=true,GenericEphemeralVolume=true,';; esac)\
```

Comment:
There's a subtle change here for 1.17: before, only `EndpointSlice=false` was set. Now `EndpointSliceProxying=false` is also set. But that feature gate was only introduced in 1.18 (https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/), so setting it for 1.17 should trigger an error.

It was disabled before for 1.19 because that's when it became enabled by default.

How about having one `$(case ${TEST_KUBERNETES_VERSION} )` with cases for one or more releases which share the same flags?

Reply:
Split setting the `EndpointSlice` and `EndpointSliceProxying` in two cases:
```
$(case ${TEST_KUBERNETES_VERSION} in 1.1[6-9]) echo 'EndpointSlice=false,';; esac)\
$(case ${TEST_KUBERNETES_VERSION} in 1.1[8-9]) echo 'EndpointSliceProxying=false,';; esac)\
```
Hopefully, it should fix the 1.17.